# Fast5 Viewer
Graphical interface for displaying fast5 files generated by pore sequencers.

Program language: 
* C++11

Compiler: 
* gcc (x86_64-posix-seh-rev0, Built by MinGW-W64 project) 8.1.0

IDE: 
* Eclipse IDE for C/C++ Developers
	* Version: 2018-09 (4.9.0)
	* Build id: 20180917-1800

Library: 
* hdf5 1.10.6

# Table of Contents

<!--ts-->
   * [fasta5 processing steps](#fasta5-processing-steps)
   * [Build Instructions](#build-instructions)
   * [License](#license)
<!--te-->

# Fast5 processing steps
| Step				| Library					| Description |
| :---					| :---						| :-- |
| Raw data acquisition			| hdf5 | Raw current data of nucleotides passing through the pore |
| Quality Control			|  				| Assess the quality of the raw current data |
| basecalling | | translating raw current data into nucleotide bases |
| alignment | | aligning nucleotide bases to reference genome |
| assembly | | trimming and detecting overlaps |
| GUI display | QT5 | visualization of raw data, base sequence, and aligned sequence |

# File Formats
| File Type				| Description					|
| :---					| :---						|
| fast5			| HDF container for Oxford Nanopore sequencer raw data |
| fastq			| base sequences with qiality score |
| SAM			| Sequence Alignment Map (SAM) is a text-based format originally for storing biological sequences aligned to a reference sequence |
| BAM			| Binary equivalent of SAM |
| PDB | PDB entries are distributed in the PDB File Format following the specification described in the Contents Guide Version 3.30 (Nov. 21, 2012). The PDB file format is no longer being modified or extended to support new content. |
| PDBx/mmCIF | Large structures (containing >62 chains and/or 99999 ATOM records) that cannot be fully represented in the PDB File Format are available in the PDB archive as single PDBx/mmCIF files |

# Databases
| Name			| Description
| :---					| :---						|
| NIH NLM SRA			| The Sequence Read Archive (SRA) stores raw sequence data and alignments	|

# Oxford Nanopore Software
| Software				| Description					|
| :---					| :---						|
| minKNOW			| Operating software that drives nanopore sequencing devices |
| Guppy			| basecalling algorithms and alignment |
| EPI2ME | cloud-based data analysis platform for QC |
| Canu | Assemebly by trimming and detecting overlaps |
| prokka | annotating the genome |
| Metrichor | integrated solutions for the whole analysis process |

# Fast5 Software Alternatives
| Software				| Language | Description					|
| :---					| :---	| :---					|
| poretools			| Python 2 | a toolkit for working with nanopore sequencing data from Oxford Nanopore |
| BulkVis			| Python 3 | enables the visualisation of raw ‘squiggle’ data from Oxford Nanopore Technologies sequencers |

# Genome Browser Software
| Software				| Language | Description					|
| :---					| :---	| :---					|
| Ensembl			| Cloud | a genome browser for vertebrate genomes that supports research in comparative genomics, evolution, sequence variation and transcriptional regulation. Ensembl annotate genes, computes multiple alignments, predicts regulatory function and collects disease data. Ensembl tools include BLAST, BLAT, BioMart and the Variant Effect Predictor (VEP) for all supported species. |

# Protein Visualization Software
| Software				| Language | Description					|
| :---					| :---	| :---					|
| BioBlender			| Python 2 | module that can be used with Blender, with the aim of providing a view of biological or molecular data |
| RasMol | C | |
| UGENE | C++ | QT5 GUI |
| UCSF Chimera | Python | |
| CLC Sequence Viewer 8.0.0 | Closed Source | proprietary |
| Geneious | Closed Source | proprietary |
| JalView | java | |
| ClustalX | C++ | |
| Alignment Annotator | javascript | |
| STRAP | JAVA | |
| PyMOL | python | |

# Build Instructions

```shell
cmake
```

# License
The library is licensed under the MIT License: <BR>

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the �Software�), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED �AS IS�, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
